### ISSUE TITLE
Dashboard error after user has been deleted

### ISSUE THREAD

**Issue by ldubrois** [SUSE/Portus#2080]
### Description

I have deleted a user from the interface, and now I no longer able to display the dashboard.

### Steps to reproduce

1. Create a user
2. Grant the user some teams
3. Delete the user

- **Expected behavior**: 
The dashboard page should be displayed without any issue
- **Actual behavior**: 

An error page is displayed. Logs from the container are :
```
Started GET "/" for 172.16.48.221 at 2018-12-27 09:06:38 +0000
Processing by DashboardController#index as HTML
Scoped order is ignored, it's forced to be batch order.
  Rendering dashboard/index.html.slim within layouts/application
  Rendered dashboard/partials/_portus_user_alert.html.slim (0.0ms)
  Rendered dashboard/partials/_stats.html.slim (6.0ms)
  Rendered public_activity/team/_remove_member.html.slim (6.0ms)
  Rendered dashboard/partials/_recent_activities.html.slim (8.5ms)
  Rendered dashboard/index.html.slim within layouts/application (15.3ms)
Completed 500 Internal Server Error in 31ms (ActiveRecord: 10.6ms)

  
ActionView::Template::Error (undefined method `display_username' for nil:NilClass):
    10:           - if activity.owner == activity.recipient
    11:             = "#{activity_owner(activity)} removed itself "
    12:           - else
    13:             = "#{activity_owner(activity)} removed user #{activity.recipient.display_username} "
    14:         = "from the "
    15:         = link_to activity_team(activity), activity.trackable
    16:         |  team
app/views/public_activity/team/_remove_member.html.slim:13:in `_app_views_public_activity_team__remove_member_html_slim__2438416966495515533_66053900'
app/views/dashboard/partials/_recent_activities.html.slim:23:in `block in _app_views_dashboard_partials__recent_activities_html_slim___3393098549991514047_61162440'
app/views/dashboard/partials/_recent_activities.html.slim:22:in `_app_views_dashboard_partials__recent_activities_html_slim___3393098549991514047_61162440'
app/views/dashboard/index.html.slim:6:in `_app_views_dashboard_index_html_slim___3108445215059580468_64745760'
Processing by ErrorsController#show as HTML
  Parameters: {"status"=>"500"}
  Rendering errors/500.html.erb within layouts/errors
  Rendered errors/500.html.erb within layouts/errors (0.1ms)
  Rendered errors/_status_title.html.slim (0.0ms)
Completed 500 Internal Server Error in 2ms (Views: 1.5ms | ActiveRecord: 0.0ms)
```

### Deployment information

**Deployment method**: 
PORTUS is deployed in a Docker Swarm cluster. 
It is a fresh new install (2 weeks ago).

**Configuration**:

You can get this information like this:

- In a container:
  - using a semi custom stack 

```yml
version: "3.2"

services:
  portus:
    image: opensuse/portus:head
    restart: always
    hostname: "{{.Service.Name}}-{{.Task.Slot}}"
    environment:
      - PORTUS_MACHINE_FQDN_VALUE=${MACHINE_FQDN}

      # DB. The password for the database should definitely not be here. You are
      # probably better off with Docker Swarm secrets.
      - PORTUS_DB_HOST=db
      - PORTUS_DB_DATABASE=portus_production
      - PORTUS_DB_PASSWORD=${DATABASE_PASSWORD}
      - PORTUS_DB_POOL=5

      # Secrets. It can possibly be handled better with Swarm's secrets.
      - PORTUS_SECRET_KEY_BASE=${SECRET_KEY_BASE}
      - PORTUS_KEY_PATH=/certificates/portus.key
      - PORTUS_PASSWORD=${PORTUS_PASSWORD}

      # SSL
      - PORTUS_CHECK_SSL_USAGE_ENABLED='false'

      # NGinx is serving the assets instead of Puma. If you want to change this,
      # uncomment this line.
      - RAILS_SERVE_STATIC_FILES='true'
    #ports:
    #  - 3000:3000
    volumes:
      - certificates:/certificates:ro
    networks:
      - vn_main
    deploy:
      mode: replicated
      replicas: 1
      labels:
        - traefik.docker.network=vn_main
        - traefik.backend=portus
        - traefik.port=3000
        - traefik.frontend.rule=Host:xxxxxxxxx

  background:
    image: opensuse/portus:head
    restart: always
    hostname: "{{.Service.Name}}-{{.Task.Slot}}"
    depends_on:
      - portus
      - db
    environment:
      # Theoretically not needed, but cconfig's been buggy on this...
      - CCONFIG_PREFIX=PORTUS
      - PORTUS_MACHINE_FQDN_VALUE=${MACHINE_FQDN}

      # DB. The password for the database should definitely not be here. You are
      # probably better off with Docker Swarm secrets.
      - PORTUS_DB_HOST=db
      - PORTUS_DB_DATABASE=portus_production
      - PORTUS_DB_PASSWORD=${DATABASE_PASSWORD}
      - PORTUS_DB_POOL=5

      # Secrets. It can possibly be handled better with Swarm's secrets.
      - PORTUS_SECRET_KEY_BASE=${SECRET_KEY_BASE}
      - PORTUS_KEY_PATH=/certificates/portus.key
      - PORTUS_PASSWORD=${PORTUS_PASSWORD}

      - PORTUS_BACKGROUND=true
    volumes:
      - certificates:/certificates:ro
    networks:
      - vn_main

  db:
    image: library/mariadb:10.0.23
    restart: always
    hostname: "{{.Service.Name}}-{{.Task.Slot}}"
    command: mysqld --character-set-server=utf8 --collation-server=utf8_unicode_ci --init-connect='SET NAMES UTF8;' --innodb-flush-log-at-trx-commit=0
    environment:
      - MYSQL_DATABASE=portus_production

      # Again, the password shouldn't be handled like this.
      - MYSQL_ROOT_PASSWORD=${DATABASE_PASSWORD}
    volumes:
      - mysql:/var/lib/mysql
    networks:
      - vn_main

  registry:
    image: library/registry:2.6
    restart: always
    hostname: "{{.Service.Name}}-{{.Task.Slot}}"
    command: ["/bin/sh", "/etc/docker/registry/init"]
    environment:
      # Authentication
      REGISTRY_AUTH_TOKEN_REALM: https://${MACHINE_FQDN}/v2/token
      REGISTRY_AUTH_TOKEN_SERVICE: ${MACHINE_FQDN}
      REGISTRY_AUTH_TOKEN_ISSUER: ${MACHINE_FQDN}
      REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE: /certificates/portus.crt

      # SSL
      #REGISTRY_HTTP_TLS_CERTIFICATE: /certificates/portus.crt
      #REGISTRY_HTTP_TLS_KEY: /certificates/portus.key

      REGISTRY_HTTP_SECRET: "xxxxxxxxxxxxxxxxx" # Some random string
      # Portus endpoint
      REGISTRY_NOTIFICATIONS_ENDPOINTS: >
        - name: portus
          url: https://${MACHINE_FQDN}/v2/webhooks/events
          timeout: 2000ms
          threshold: 5
          backoff: 1s
    volumes:
      - registry_data:/var/lib/registry
      - certificates:/certificates:ro
      - /xxxxxxxxxxx/config.yml:/etc/docker/registry/config.yml:ro # Moche
      - /xxxxxxxxxxx/init:ro # Moche
    networks:
      - vn_main
    deploy:
      mode: replicated
      replicas: 1
      labels:
        - traefik.docker.network=vn_main
        - traefik.backend=registry
        - traefik.port=5000
        - traefik.frontend.rule=Host:xxxxxxxx
    #ports:
    #  - 5000:5000
    #  - 5001:5001 # required to access debug service

networks:
  vn_main:
    external: true


volumes:
  certificates:
    xxxxxxxx
  mysql:
    xxxxxxxx
  registry_data:
    xxxxxxxx

```

**Portus version**: 
Version: 2.5.0-dev@35dbf226e88506394cc608217bada0ca62cd220d

**Comment by vitoravelino** [comment]
Indeed. Thanks for reporting it!

**Comment by vitoravelino** [comment]
Fixed in #2081
